### PR TITLE
Issue #13411: Fix NPE in UnnecessarySemicolonAfterTypeMemberDeclarationCheck

### LINKED ISSUE #13411: UnnecessarySemicolonAfterTypeMemberDeclaration throws Exception on on Local Class
I have read check documentation: https://checkstyle.sourceforge.io/checks/coding/unnecessarysemicolonaftertypememberdeclaration.html#UnnecessarySemicolonAfterTypeMemberDeclaration
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
➜  src javac SwitchScope.java 

➜  src cat SwitchScope.java 
public class SwitchScope {
    public static void main(String[] args) {
        switch (args.length) {
        case 0:
            final int k = 12;
            class Local {
                int j = k;
            }
        case 1:
        }
    }
}
                                                                                                           
➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="UnnecessarySemicolonAfterTypeMemberDeclaration"/>
 </module>
</module>

➜  src java -jar checkstyle-10.12.1-all.jar -c config.xml SwitchScope.java 
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing SwitchScope.java
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:306)
        at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:223)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:415)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:338)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:195)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:130)
Caused by: java.lang.NullPointerException: Cannot invoke "com.puppycrawl.tools.checkstyle.api.DetailAST.getType()" because "ast" is null
        at com.puppycrawl.tools.checkstyle.checks.coding.UnnecessarySemicolonAfterTypeMemberDeclarationCheck.isSemicolon(UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java:236)
        at com.puppycrawl.tools.checkstyle.checks.coding.UnnecessarySemicolonAfterTypeMemberDeclarationCheck.checkTypeDefinition(UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java:196)
        at com.puppycrawl.tools.checkstyle.checks.coding.UnnecessarySemicolonAfterTypeMemberDeclarationCheck.visitToken(UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java:165)
        at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:335)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:406)
        at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:273)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:154)
        at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:98)
        at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:334)
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:293)
        ... 5 more
Checkstyle ends with 1 errors.

```

---

**Describe what you expect in detail.**

I would expect this check not to throw a Null Pointer Exception on a local class declaration within a switch statement, even if this code is unusual.

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Issue #13411: Fix NPE in UnnecessarySemicolonAfterTypeMemberDeclarationCheck

Diff Regression config: https://gist.githubusercontent.com/relentless-pursuit/95180658ff45ac1bab444f2c35ac301d/raw/d53eda164c7ccd456cb6a1def8bb4e4f0de9c39c/gistfile1.txt

### CODE REVIEW

**Comment by relentless-pursuit** [comment]
@romani and @nrmancuso 

Sorry for the hassle. This is a new PR. Kindly review whenever convenient for you. I have made changes as requested by @nrmancuso. 

Past PR's conversation as a reference:https://github.com/checkstyle/checkstyle/pull/13417

**Comment by romani** [comment]
Please fix https://github.com/checkstyle/checkstyle/actions/runs/5708338178/job/15466147589?pr=13486#step:6:14

**Comment by relentless-pursuit** [comment]
> Please fix https://github.com/checkstyle/checkstyle/actions/runs/5708338178/job/15466147589?pr=13486#step:6:14

done. can i be assigned reviewers as i have made requested changes?

**Comment by romani** [comment]
Please do report generation https://github.com/checkstyle/checkstyle/pull/13417#pullrequestreview-1551157067

Examples: https://github.com/checkstyle/checkstyle/pull/13475

**Comment by relentless-pursuit** [comment]
Github, generate report

**Comment by relentless-pursuit** [comment]
> Please do report generation [#13417 (review)](https://github.com/checkstyle/checkstyle/pull/13417#pullrequestreview-1551157067)
> 
> Examples how others are doing it: #13475

Bit confusing. Reading the documentation. Will seek help on this.

**Comment by relentless-pursuit** [comment]
GitHub, generate report

**Comment by relentless-pursuit** [comment]
> Resolve #13411
> 
> Diff Regression config: https://gist.githubusercontent.com/relentless-pursuit/95180658ff45ac1bab444f2c35ac301d/raw/d53eda164c7ccd456cb6a1def8bb4e4f0de9c39c/gistfile1.txt Diff Regression projects: https://gist.githubusercontent.com/relentless-pursuit/b51c223730e8e119e34855226bef3865/raw/6989794a8eb9ad5fde3995f637617cc775173973/gistfile1.txt

@Romani, have i done it right?

**Comment by relentless-pursuit** [comment]
GitHub, generate report

**Comment by nrmancuso** [comment]
@relentless-pursuit click through link at https://github.com/checkstyle/checkstyle/pull/13486#issuecomment-1660361711 and read error messages. Also - you do not need to use a projects file, you can remove that from the PR description.

**Comment by relentless-pursuit** [comment]
GitHub, generate report

**Comment by relentless-pursuit** [comment]
> @relentless-pursuit please rebase on latest master and resolve conflicts to prepare for final review, report looks good

done.

File: config/projects-to-test/openjdk20-excluded.files
Comment (nrmancuso) [inline config/projects-to-test/openjdk20-excluded.files:1]: The only change in this file should be the removal of related suppression module, we should not add modules back.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: config/projects-to-test/openjdk20-excluded.files
Comment (relentless-pursuit) [reply config/projects-to-test/openjdk20-excluded.files:1]: > The only change in this file should be the removal of related suppression module, we should not add modules back.

the recent push looks fine. please let me know your feedback.

Comment (relentless-pursuit) [reply config/projects-to-test/openjdk20-excluded.files:1]:
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':postgresql'.
> Could not determine the dependencies of null.
   > Could not resolve all task dependencies for configuration ':postgresql:classpath'.
      > Could not resolve gradle.plugin.com.github.lburgazzoli:gradle-karaf-plugin:0.5.6.
        Required by:
            project :postgresql > com.github.lburgazzoli.karaf:com.github.lburgazzoli.karaf.gradle.plugin:0.5.6
         > Could not resolve gradle.plugin.com.github.lburgazzoli:gradle-karaf-plugin:0.5.6.
            > Could not get resource 'https://plugins.gradle.org/m2/gradle/plugin/com/github/lburgazzoli/gradle-karaf-plugin/0.5.6/gradle-karaf-plugin-0.5.6.pom'.
               > Could not GET 'https://plugins.gradle.org/m2/gradle/plugin/com/github/lburgazzoli/gradle-karaf-plugin/0.5.6/gradle-karaf-plugin-0.5.6.pom'. Received status code 530 from server: 
```

Not sure why it is failing.

**Comment by relentless-pursuit** [comment]
GitHub, generate report

File: config/projects-to-test/openjdk20-excluded.files
Comment (nrmancuso) [reply config/projects-to-test/openjdk20-excluded.files:1]: Latest push looks good.

**Pull request merged**